erpnext/accounts/doctype/payment_request/test_payment_request.py

```diff
@@ -75,6 +108,83 @@ def test_payment_request_linkings(self):
 		self.assertEqual(pr.reference_name, si_usd.name)
 		self.assertEqual(pr.currency, "USD")
 
+	def test_payment_channels(self):
+		so = make_sales_order(currency="USD")
+
+		pr = make_payment_request(
+			dt="Sales Order",
+			dn=so.name,
+			payment_gateway_account="_Test Gateway - USD",  # email channel
+			submit_doc=False,
+			return_doc=True,
+		)
+		pr.flags.mute_email = True  # but temporarily prohibit sending
+		pr.submit()
+		pr.reload()
+		self.assertEqual(pr.payment_channel, "Email")
+		self.assertEqual(pr.mute_email, False)
+
+		self.assertIsNone(pr.payment_url)
+		self.assertEqual(self.send_email.call_count, 0)  # hence: no increment
+		self.assertEqual(self._get_payment_gateway_controller.call_count, 1)
+		pr.cancel()
+
+		pr = make_payment_request(
+			dt="Sales Order",
+			dn=so.name,
+			payment_gateway_account="_Test Gateway Phone - USD",
+			submit_doc=True,
+			return_doc=True,
+		)
+		pr.reload()
+
+		self.assertEqual(pr.payment_channel, "Phone")
+		self.assertEqual(pr.mute_email, False)
+
+		self.assertIsNone(pr.payment_url)
+		self.assertEqual(self.send_email.call_count, 0)  # no increment on phone channel
+		self.assertEqual(self._get_payment_gateway_controller.call_count, 3)
+		pr.cancel()
+
+		pr = make_payment_request(
+			dt="Sales Order",
+			dn=so.name,
+			payment_gateway_account="_Test Gateway - USD",  # email channel
+			submit_doc=True,
+			return_doc=True,
+		)
+		pr.reload()
+
+		self.assertEqual(pr.payment_channel, "Email")
+		self.assertEqual(pr.mute_email, False)
+
+		self.assertEqual(pr.payment_url, PAYMENT_URL)
+		self.assertEqual(self.send_email.call_count, 1)  # increment on normal email channel
+		self.assertEqual(self._get_payment_gateway_controller.call_count, 4)
+		pr.cancel()
+
+		so = make_sales_order(currency="USD", do_not_save=True)
+		# no-op; for optical consistency with how a webshop SO would look like
+		so.order_type = "Shopping Cart"
+		so.save()
+		pr = make_payment_request(
+			dt="Sales Order",
+			dn=so.name,
+			payment_gateway_account="_Test Gateway - USD",  # email channel
+			order_type="Shopping Cart",
+			submit_doc=True,
+			return_doc=True,
+		)
+		pr.reload()
+
+		self.assertEqual(pr.payment_channel, "Email")
+		self.assertEqual(pr.mute_email, False)
+
+		self.assertIsNone(pr.payment_url)
+		self.assertEqual(self.send_email.call_count, 1)  # no increment on shopping cart
+		self.assertEqual(self._get_payment_gateway_controller.call_count, 5)
```

Comment:
Each test would hit it once, I just wanted to make sure of that as an invariant to the tests so that on later improvements (e.g. when enabling a dispatcher between legacy and new-style gateway) we would be very deliberate in the test cases about reflecting that.